hail/src/main/scala/is/hail/types/physical/PNDArray.scala

```diff
@@ -54,7 +56,11 @@ abstract class PNDArray extends PType {
 abstract class PNDArrayValue extends PValue {
   def apply(indices: IndexedSeq[Value[Long]], mb: EmitMethodBuilder[_]): Value[_]
 
+  override def pt: PNDArray = ???
```

Comment:
Why not leave abstract?

Reply:
This lets me set the return type for things that override it so they don't have to be cast.

Reply:
Tim and I previously handled this using 

```scala
private def ptND: PNDArray
override def pt: PNDArray = ptND
```